libs/core/langchain_core/messages/utils.py

```diff
@@ -242,3 +255,136 @@ def convert_to_messages(
         List of messages (BaseMessages).
     """
     return [_convert_to_message(m) for m in messages]
+
+
+def filter_messages(
+    messages: Sequence[MessageLikeRepresentation],
+    *,
+    incl_names: Optional[Sequence[str]] = None,
+    excl_names: Optional[Sequence[str]] = None,
+    incl_types: Optional[Sequence[Union[str, Type[BaseMessage]]]] = None,
+    excl_types: Optional[Sequence[Union[str, Type[BaseMessage]]]] = None,
+    incl_ids: Optional[Sequence[str]] = None,
+    excl_ids: Optional[Sequence[str]] = None,
+) -> List[BaseMessage]:
+    if incl_names and excl_names:
+        raise ValueError
+    if incl_types and excl_types:
+        raise ValueError
+    if incl_ids and excl_ids:
+        raise ValueError
+    messages = convert_to_messages(messages)
+    incl_types_str = [t for t in (incl_types or ()) if isinstance(t, str)]
+    incl_types_types = tuple(t for t in (incl_types or ()) if isinstance(t, type))
+    excl_types_str = [t for t in (excl_types or ()) if isinstance(t, str)]
+    excl_types_types = tuple(t for t in (excl_types or ()) if isinstance(t, type))
+
+    filtered: List[BaseMessage] = []
+    for msg in messages:
+        if incl_names and msg.name not in incl_names:
+            continue
+        elif excl_names and msg.name in excl_names:
+            continue
+        elif incl_types_str and msg.type not in incl_types_str:
+            continue
+        elif incl_types_types and not isinstance(msg, incl_types_types):
+            continue
+        elif excl_types_str and msg.type in excl_types_str:
+            continue
+        elif excl_types_types and isinstance(msg.type, excl_types_types):
+            continue
+        elif incl_ids and msg.id not in incl_ids:
+            continue
+        elif excl_ids and msg.id in excl_ids:
+            continue
+        else:
+            filtered.append(msg)
+    return filtered
+
+
+def abbreviate_messages(
```

Comment:
Trim sounds better to me even if there's summarization

Reply:
personally disagree (trim really sounds like just shaving off some, not condensing/summarizing) but dont feel strongly